includes/create-theme/theme-locale.php

```diff
@@ -28,6 +28,72 @@ private static function escape_text_content( $string ) {
 
 		$string = addcslashes( $string, "'" );
 
+		// Process the string to avoid escaping inner HTML markup.
+		$p = new WP_HTML_Tag_Processor( $string );
+
+		$text   = '';
+		$tokens = array();
+		while ( $p->next_token() ) {
+			$token_type    = $p->get_token_type();
+			$token_name    = strtolower( $p->get_token_name() );
+			$is_tag_closer = $p->is_tag_closer();
+
+			if ( '#tag' === $token_type ) {
+				// Add a placeholder for the token.
+				$text .= '%s';
+				if ( $is_tag_closer ) {
+					$tokens[] = "</{$token_name}>";
+				} else {
+					// Depending on the HTML tag, we may need to process attributes so they are correctly added to the placeholder.
+					switch ( $token_name ) {
```

Comment:
One thing I would like feedback on is whether this is too fragile, complicated, or tedious an approach to maintain.

I plan to add a few more unit tests.

Reply:
I think we're good to rely on the output of `get_token_name()`, so based on that, I believe this is a fine way to parse these tags. We have a limited scope since we're currently only dealing with `a`, `img` and `mark`, and otherwise we default to the original tag.

That's only my opinion though 😅 Was there anything you were specifically concerned about with this approach?

Reply:
I think just needing to add cases for specific tags and if the attributes changes, it may break. 

But the options for adding inline HTML / formatting inside blocks are rather limited, so maybe it's okay.

Reply:
Do we have to deconstruct the attributes and reassemble them? I'm worried about dropping attributes or other parts of the markup that might be added by plugins or filters.

For example, could we replace `<a href="https://wordpress.org">WordPress</a>` with something like
```php
<a href="<?php echo esc_url( 'https://wordpress.org' ) >?"><?php echo esc_html__( 'WordPress' ) ?></a>
``` 
but maintain any other attributes added to the `<a>` tag?

I'm not really worried about escaping every possible attribute, I think the main thing is to make sure the translated strings are escaped as a form of user input.

Reply:
> I'm worried about dropping attributes or other parts of the markup that might be added by plugins or filters.

That's a good point. I could not find a way with the HTML Tag Processor to just carry over the whole tag token and all its attributes. But I refactored the approach so that all the attributes should be carried over, rather than processing the attributes based on the type of tag: https://github.com/WordPress/create-block-theme/pull/719/commits/678a8ef22ddecc2e0b4190dd141cfa0c4436cd1e#diff-fc11ad66397ed68725a8fe74f3031f104368f850b5b94370123482ec9e719e1bR54-R70